### PR TITLE
[Bug] Correct Weight Offloader's Attribute Name for torch.nn.Parameter

### PR DESCRIPTION
**Pull request by xiaobao520123**
## Motivation

This PR is part of source PR: https://github.com/sgl-project/sglang/pull/24531.

Correct the attribute name of `weight_loader` when checking if parameters have.

## Modifications

1. Correct the attribute's name. `weihgt_loader` -> `weight_loader`.

## Accuracy Tests

Please refer to source PR: https://github.com/sgl-project/sglang/pull/24531.

**Result**:

- Qwen3-8B

Configuration | Score | Latency (s) | Throughput (tok/s)
-- | -- | -- | --
Baseline | 0.930 | 55.71 | 2707.19
With offloading | 0.935 | 358.26 | 422.47

## Speed Tests and Profiling

Please refer to source PR: https://github.com/sgl-project/sglang/pull/24531.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :hourglass_flowing_sand: [Run #26104515010](https://github.com/sgl-project/sglang/actions/runs/26104515010)
Latest PR Test (Extra): :warning: **Not enabled** -- add `run-ci-extra` label to opt in.

### CODE REVIEW

**Review by fzyzcjy — APPROVED** [review]

**Comment by fzyzcjy** [comment]
LGTM given old code is surely a typo; but need to prove testing correctness before merging

**Comment by xiaobao520123** [comment]
> LGTM given old code is surely a typo; but need to prove testing correctness before merging

@fzyzcjy, the correctness, accuracy, and profiling test are done in source PR: https://github.com/sgl-project/sglang/pull/24531. Please check the report there. Thanks!

**Review by ch-wan — APPROVED** [review]
This PR passes all tests for param offloading. I'm going to merge it.
